### ISSUE TITLE
Deployment Improvements

### ISSUE THREAD

**Issue by ekingery** [exercism/cli#893]
This issue groups a set of related updates and features we'd like to implement now that we are using GoReleaser. I am open to breaking any of these into individual tickets, particularly when someone is ready to work on them. I think storing them in one place for now is probably best though. 

 - [ ] disable auto-upgrade functionality if installed by package manager (win, mac, linux)
   - [ ] improve the upgrade messaging - right now nothing happens when you upgrade, we should give the user a notification of successful upgrade
  - [ ] Weigh [auto generation of changelog](https://goreleaser.com/customization/#customize-the-changelog) vs our customizations (not including refactors, test updates, etc) (also [the release notes](https://goreleaser.com/customization/#custom-release-notes)
 - [ ] Replace hardcoded `Version` constant in cmd/version.go with [main.version](https://goreleaser.com/environment/#using-the-main-version) - [SO post on injecting to cmd/version.go](https://stackoverflow.com/a/47510909)
- [ ] Investigate [Flatpak](https://flatpak.org/) as a secondary distribution method for Linux
- [ ] Consider generating .deb and .rpm using nfpm: https://goreleaser.com/nfpm/
- [ ] Use [GoReleaser CI features](https://goreleaser.com/ci/) and/or Github Actions [GoReleaser Actions](https://goreleaser.com/actions) to automate some / all of the release process
  - [ ] also consider [Snap auto-deploys](https://snapcraft.io/build)

**Comment by kytrinyx** [comment]
> Weigh auto generation of changelog vs our customizations

I feel pretty strongly about manually creating a change log, and not including refactors and test fixes etc. I find automated changelogs to be hard to parse mentally, as I don't know what was important and not.

**Comment by ekingery** [comment]
> > Weigh auto generation of changelog vs our customizations
> 
> I feel pretty strongly about manually creating a change log, and not including refactors and test fixes etc. I find automated changelogs to be hard to parse mentally, as I don't know what was important and not.

Roger that. I think a good goal would be to be able to generate the changelog automatically according to our preferred formatting, possibly using regex exclude filters for tests and refactors, and then we can have a person review and edit before committing it. If it's not saving time / effort or meeting our standards, we can ditch the automation. Also, at our current pace, it's probably not really worth the automation effort anyway.

**Comment by kytrinyx** [comment]
> I think a good goal would be to be able to generate the changelog automatically according to our preferred formatting, possibly using regex exclude filters for tests and refactors, and then we can have a person review and edit before committing. 

I like that. I wonder if we could have a simple tool that gets the current sha, the sha of the most recent release, and lists all the PRs in our desired format, then we can copy/paste and delete the things we don't care about.

**Comment by shashank-priyadarshi** [comment]
@ekingery can I please take this up?

**Comment by ekingery** [comment]
> @ekingery can I please take this up?

I defer to @ErikSchierboom on whether any of these changes are still compatible with the future direction of the CLI. If you don't hear back, I would not spend time on these suggestions as they are quite out of date.

**Comment by ErikSchierboom** [comment]
We're using GoReleaser CI and Actions right now, so that is fixed. Auto-generation of the release notes, I'm not entirely sure what the status is there. The others to me sound like less of a big deal?

**Comment by ekingery** [comment]
Thanks Erik. I'll go ahead and close the issue - we can always open up individual tickets for any of these that become a priority.
